test/_unit/apps/collection/validation/validation.spec.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const Controller = require('hof').controller;
+
+const DummyController = class DummyControllerBase extends Controller {
+  configure(req, res, next) {
+    super.configure(req, res, next);
+  }
+};
+describe('apps/collection/fields/application-type', () => {
+  let controller;
+  let req;
+  let res;
+
+  beforeEach(() => {
+    req = reqres.req();
+    res = reqres.res();
+
+    req.form.options = {};
+
+    controller = new DummyController({ template: 'index', route: '/index' });
+  });
+
+  describe('validation', () => {
+    let sandbox;
+
+    beforeEach(() => {
+      sandbox = sinon.createSandbox();
+      req.form.options.fields = {
+        'application-type': {
+          validate: ['required']
+        },
+        'application-type-other': {
+          validate: ['required']
+        }
+      };
+    });
+
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it('adds a `required` validator if the application type is not selected', () => {
+      req.sessionModel.set('application-type');
+      controller.configure(req, res, () => {
+        req.form.options.fields['application-type'].should.have.property('validate');
+        expect(req.form.options.fields['application-type'].validate).to.deep.eql(['required' ]);
+      });
+    });
+
+    it('adds a`required` validator to the `other` input box', () => {
+      req.sessionModel.set('application-type', 'other');
+      controller.configure(req, res, () => {
+        req.form.options.fields['application-type-other'].should.have.property('validate');
+        expect(req.form.options.fields['application-type-other'].validate).to.deep.eql([ 'required']);
+      });
+    });
+  });
+});
```

Comment:
oooh nice work @Rhodine I prefer these tests rather than lots of acceptance tests which are slow and cause build issues ⭐